site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -146,69 +184,251 @@ def _transform_publication_date(self):
                 if first_name and last_name:
-                    rdm_creator["person_or_org"]["name"] = (
+                    rdm_creatibutor["person_or_org"]["name"] = (
                         author.get("last_name") + ", " + author.get("first_name")
                     )
-                creators.append(rdm_creator)
-            return creators
+
+                creator_affiliations = self._transform_author_affiliations(author)
+                creator_identifiers = self._transform_author_identifiers(author)
+                role = author.get("inspire_roles")
+
+                if creator_affiliations:
+                    rdm_creatibutor["affiliations"] = creator_affiliations
+
+                if creator_identifiers:
+                    rdm_creatibutor["person_or_org"][
+                        "identifiers"
+                    ] = creator_identifiers
+
+                if role:
+                    rdm_creatibutor["role"] = role[0]
+                creatibutors.append(rdm_creatibutor)
+            return creatibutors
         except Exception as e:
             self.metadata_errors.append(
                 f"Error occurred while mapping INSPIRE authors to creators and contributors. INSPIRE "
                 f"record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
             )
             return None
 
+    def _transform_author_identifiers(self, author):
+        """Transform ids of authors. Keeping only ORCID and CDS."""
+        author_ids = author.get("ids", [])
+        processed_identifiers = []
+
+        schemes_map = {
+            "INSPIRE ID": "inspire_author",
+            "ORCID": "orcid",
+        }
+
+        for author_id in author_ids:
+            author_schema = author_id.get("schema")
+            if author_schema in schemes_map.keys():
+                processed_identifiers.append(
+                    {
+                        "identifier": author_id.get("value"),
+                        "scheme": schemes_map[author_schema],
+                    }
+                )
+
+        return processed_identifiers
+
+    def _transform_author_affiliations(self, author):
+        """Transform affiliations."""
+        affiliations = author.get("affiliations", [])
+        mapped_affiliations = []
+
+        for affiliation in affiliations:
+            value = affiliation.get("value")
+            if value:
+                mapped_affiliations.append({"name": value})
+
+        return mapped_affiliations
+
+    def _transform_copyrights(self):
+        """Transform copyrights."""
+        # format: "© {holder} {year}, {statement} {url}"
+        copyrights = self.inspire_metadata.get("copyright", [])
+        result_list = []
+
+        for cp in copyrights:
+            holder = cp.get("holder", "")
+            statement = cp.get("statement", "")
+            url = cp.get("url", "")
+            year = str(cp.get("year", ""))
```

Comment:
how will the entry look like (rest API) if we have only statement and the rest of values empty? will it assign empty strings?

Reply:
It will look like this: "© All rights reserved" provided `statement = "All rights reserved"`